components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/CacheBackedOrganizationManagementDAO.java

```diff
@@ -0,0 +1,418 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.organization.management.service.dao.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.organization.management.service.cache.OrganizationDetailsCacheByOrgId;
+import org.wso2.carbon.identity.organization.management.service.cache.OrganizationDetailsCacheEntry;
+import org.wso2.carbon.identity.organization.management.service.cache.OrganizationIdCacheKey;
+import org.wso2.carbon.identity.organization.management.service.cache.TenantDomainCacheByOrgId;
+import org.wso2.carbon.identity.organization.management.service.cache.TenantDomainCacheEntry;
+import org.wso2.carbon.identity.organization.management.service.dao.OrganizationManagementDAO;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementServerException;
+import org.wso2.carbon.identity.organization.management.service.filter.ExpressionNode;
+import org.wso2.carbon.identity.organization.management.service.model.BasicOrganization;
+import org.wso2.carbon.identity.organization.management.service.model.Organization;
+import org.wso2.carbon.identity.organization.management.service.model.PatchOperation;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+import static org.wso2.carbon.identity.organization.management.service.constant.OrganizationManagementConstants.DEFAULT_ORGANIZATION_DEPTH_IN_HIERARCHY;
+import static org.wso2.carbon.identity.organization.management.service.constant.OrganizationManagementConstants.SUPER_ORG_ID;
+
+/**
+ * Cached DAO of organization management. All the DAO access should happen through this layer to ensure single point
+ * of caching.
+ */
+public class CacheBackedOrganizationManagementDAO implements OrganizationManagementDAO {
+
+    private final OrganizationManagementDAO organizationMgtDAO;
+
+    public CacheBackedOrganizationManagementDAO(OrganizationManagementDAO organizationMgtDAO) {
+
+        this.organizationMgtDAO = organizationMgtDAO;
+    }
+
+    @Override
+    public void addOrganization(Organization organization) throws OrganizationManagementServerException {
+
+        organizationMgtDAO.addOrganization(organization);
+    }
+
+    @Override
+    public boolean isOrganizationExistByName(String organizationName) throws OrganizationManagementServerException {
+
+        return organizationMgtDAO.isOrganizationExistByName(organizationName);
+    }
+
+    @Override
+    public boolean isOrganizationExistById(String organizationId) throws OrganizationManagementServerException {
+
+        String tenantDomain = resolveTenantDomain(organizationId);
+        if (tenantDomain == null) {
+            return organizationMgtDAO.isOrganizationExistById(organizationId);
+        }
```

Comment:
since you return here and close if condition, you can add an empty line after this to improve the readability. follow the same for other places.